shell/shared/grain.js

```diff
@@ -540,12 +546,58 @@ if (Meteor.isClient) {
     },
   });
 
+  Template.shareWithOthers.onRendered(function () {
+    this.find("[role=tab]").focus();
+  });
+  var activateTargetTab = function(event, instance) {
+    // Deactivate all tabs and all tab panels.
+    instance.findAll("ul[role=tablist]>li[role=tab]").forEach(function (element) {
+      element.setAttribute("aria-selected", false);
+    });
+    instance.findAll(".tabpanel").forEach(function (element) {
+      element.setAttribute("aria-hidden", true);
+    });
+
+    // Activate the tab header the user selected.
+    event.currentTarget.setAttribute("aria-selected", true);
+    // Show the corresponding tab panel.
+    var idToShow = event.currentTarget.getAttribute("aria-controls");
+    var tabPanelToShow = instance.find("#" + idToShow);
+    tabPanelToShow.setAttribute("aria-hidden", false);
+  };
   Template.shareWithOthers.events({
-    "click .shareable-link": function (event, instance) {
-      instance.find(".share-tabs").setAttribute("data-which-tab", "shareable-link");
-    },
-    "click .send-invite": function (event, instance) {
-      instance.find(".share-tabs").setAttribute("data-which-tab", "send-invite");
+    "click #send-invite-tab-header": activateTargetTab,
+    "click #shareable-link-tab-header": activateTargetTab,
+    "keydown [role=tab]": function(event, template) {
+      if (event.keyCode == 38 || event.keyCode == 40) { // up and down arrows
```

Comment:
It seems sad that we have to write all of this special keyboard handling logic. I suppose there's no betters way to make it easy to focus on either tab?